go/dpagg/mean.go

```diff
@@ -213,6 +216,102 @@ func (bm *BoundedMeanFloat64) Result() float64 {
 	return clamped
 }
 
+// ComputeConfidenceInterval computes a confidence interval that contains the true mean with a probability
+// greater than or equal to 1 - alpha using the confidence interval for sum and count.
+//
+// Note Result() needs to be called before ComputeConfidenceInterval, otherwise this will return an error.
+func (bm *BoundedMeanFloat64) ComputeConfidenceInterval(alpha float64) (noise.ConfidenceInterval, error) {
+	if !bm.resultReturned {
+		return noise.ConfidenceInterval{}, fmt.Errorf("You need to call Result() before calling ComputeConfidenceInterval()")
+	}
+	// Brute force for the tightest confidence interval by iterating through
+	// a fixed number of splits numSplits for countAlpha where it takes the
+	// range of values strictly from 0 to alpha.
+	increment := alpha / float64(numSplits)
+	curCountAlpha := increment
+	bestTightness := math.Inf(1)
+	var bestConfInt noise.ConfidenceInterval
+	for curCountAlpha < alpha {
+		confInt, err := bm.computeConfidenceIntervalSplit(alpha, curCountAlpha)
+		if err != nil {
+			return noise.ConfidenceInterval{}, err
+		}
+		tightness := confInt.UpperBound - confInt.LowerBound
+		if tightness < bestTightness {
+			bestTightness = tightness
+			bestConfInt = confInt
+		}
+		curCountAlpha += increment
+	}
+	return bestConfInt, nil
+}
+
+// As opposed to the former function, this function has countAlpha where you can set confidence
+// level for count confidence interval. It computes the coresponding alpha for sum confidence
+// interval to return a confidence interval with a confidence level of at least 1 - meanAlpha.
```

Comment:
Maybe something like `computeConfidenceIntervalSplit computes a confidence interval for mean with a given 1 - countAlpha confidence level for count. It computes the corresponding alpha for sum that returns a confidence interval with a confidence level of at least 1 - meanAlpha.`